### PR TITLE
[Fix] Fixed mypy warnings

### LINKED ISSUE #1894: Errors in mypy test locally
### Describe the bug

When run locally 
`make mypy`

I am getting many errors from augmentation package

```

===================================================================== FAILURES ======================================================================
___________________________________________________________________ test session ____________________________________________________________________
mypy exited with status 1.
________________________________________________________ kornia/augmentation/_2d/mix/base.py ________________________________________________________
252: error: Dict entry 0 has incompatible
252: error: Dict entry 1 has incompatible
_______________________________________________________ kornia/augmentation/_2d/mix/cutmix.py _______________________________________________________
225: error: No overload variant of "to"
225: note: Possible overload variants:
225: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
225: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
225: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
226: error: No overload variant of "to"
226: note: Possible overload variants:
226: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
226: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
226: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
239: error: No overload variant of
239: note: Possible overload variants:
239: note:     def zeros(size: Union[Size, List[int], Tuple[int, ...]], *, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
239: note:     def zeros(*size: int, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
239: note:     def zeros(size: Sequence[Union[int, SymInt]], *, out: Optional[Tensor] = ..., dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
239: note:     def zeros(*size: int, out: Optional[Tensor] = ..., dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
240: error: Item "None" of
240: error: Argument 1 to "range" has
244: error: No overload variant of "to"
244: note: Possible overload variants:
244: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
244: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
244: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
245: error: No overload variant of "to"
245: note: Possible overload variants:
245: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
245: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
245: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
_______________________________________________________ kornia/augmentation/_2d/mix/mixup.py ________________________________________________________
205: error: No overload variant of "to" of
205: note: Possible overload variants:
205: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
205: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
205: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
206: error: No overload variant of "to" of
206: note: Possible overload variants:
206: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
206: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
206: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
207: error: No overload variant of "zeros"
207: note: Possible overload variants:
207: note:     def zeros(size: Union[Size, List[int], Tuple[int, ...]], *, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
207: note:     def zeros(*size: int, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
207: note:     def zeros(size: Sequence[Union[int, SymInt]], *, out: Optional[Tensor] = ..., dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
207: note:     def zeros(*size: int, out: Optional[Tensor] = ..., dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
220: error: No overload variant of "to" of
220: note: Possible overload variants:
220: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
220: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
220: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
221: error: No overload variant of "to" of
221: note: Possible overload variants:
221: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
221: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
221: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
222: error: No overload variant of "to" of
222: note: Possible overload variants:
222: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
222: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
222: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
___________________________________________________ kornia/augmentation/random_generator/base.py ____________________________________________________
38: error: Incompatible types in
39: error: Incompatible types in
________________________________________________ kornia/augmentation/random_generator/_2d/resize.py _________________________________________________
35: error: Incompatible
36: error: Incompatible
======================================================================= mypy ========================================================================
assignment (expression has type "device", variable has type "None") 
[assignment]
                self.device = device
                              ^
assignment (expression has type "dtype", variable has type "None")  [assignment]
                self.dtype = dtype
                             ^
types in assignment (expression has type "Optional[device]", base class
"RandomGeneratorBase" defined the type as "None")  [assignment]
            self.device: Optional[torch.device] = None
                                                  ^
types in assignment (expression has type "Optional[dtype]", base class
"RandomGeneratorBase" defined the type as "None")  [assignment]
            self.dtype: Optional[torch.dtype] = None
                                                ^
type "str": "device"; expected "str": "Tensor"  [dict-item]
                self._params.update({"device": in_tensor.device, "dtype": ...
                                     ^
type "str": "dtype"; expected "str": "Tensor"  [dict-item]
    ...self._params.update({"device": in_tensor.device, "dtype": in_tensor.dt...
                                                        ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    input.to(device=input.device, dtype=params["dtype"]),
                    ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    input.to(device=input.device, dtype=params["dtype"]),
                    ^
matches argument types "Tuple[int]", "device", "Tensor"  [call-overload]
                    torch.zeros((len(input),), device=input.device, dtype=...
                    ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    input.to(device=input.device, dtype=params["dtype"]),
                    ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    labels_permute.to(device=input.device, dtype=params["d...
                    ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    params["mixup_lambdas"].to(device=input.device, dtype=...
                    ^
of "_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                            input.to(device=input.device, dtype=params["dt...
                            ^
of "_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                            labels_permute.to(device=input.device, dtype=p...
                            ^
"zeros" matches argument types "int", "device", "Tensor"  [call-overload]
            lam = torch.zeros((len(input)), device=input.device, dtype=par...
                  ^
"Optional[RandomGeneratorBase]" has no attribute "num_mix"  [union-attr]
            for _ in range(self._param_generator.num_mix):
                           ^
incompatible type "Union[Tensor, Module, Any]"; expected "SupportsIndex" 
[arg-type]
            for _ in range(self._param_generator.num_mix):
                           ^
of "_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                            input.to(device=input.device, dtype=params["dt...
                            ^
of "_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                            input.to(device=input.device, dtype=params["dt...
                            ^
Found 19 errors in 5 files (checked 272 source files)
============================================================== short test summary info ==============================================================
FAILED kornia/__init__.py::mypy-status
FAILED kornia/augmentation/_2d/mix/base.py::mypy
FAILED kornia/augmentation/_2d/mix/cutmix.py::mypy
FAILED kornia/augmentation/_2d/mix/mixup.py::mypy
FAILED kornia/augmentation/random_generator/base.py::mypy
FAILED kornia/augmentation/random_generator/_2d/resize.py::mypy

```

### Reproduction steps

```bash
make mypy
```

### Expected behavior

No error

### Environment

```shell
Versions

Collecting environment information...
PyTorch version: 1.13.0.dev20220916
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.6 (arm64)
GCC version: Could not collect
Clang version: 14.0.6
CMake version: version 3.22.3
Libc version: N/A

Python version: 3.9.4 | packaged by conda-forge | (default, May 10 2021, 22:10:52) [Clang 11.1.0 ] (64-bit runtime)
Python platform: macOS-12.6-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy==0.910
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.2
[pip3] pytest-mypy==0.9.0
[pip3] torch==1.13.0.dev20220916
[pip3] torch-dimcheck==0.0.1
[pip3] torchaudio==0.11.0
[pip3] torchvision==0.10.0
[conda] numpy 1.21.2 pypi_0 pypi
[conda] torch 1.13.0.dev20220916 pypi_0 pypi
[conda] torch-dimcheck 0.0.1 pypi_0 pypi
[conda] torchaudio 0.11.0 pypi_0 pypi
[conda] torchvision 0.10.0 pypi_0 pypi
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shijianjian**
partials https://github.com/kornia/kornia/issues/1894

### CODE REVIEW

File: kornia/constants.py [hunk 1]
```diff
@@ -53,6 +54,42 @@ class SamplePadding(ConstantBase, Enum, metaclass=EnumMetaFlags):
...
 
 
+class DType(ConstantBase, Enum, metaclass=EnumMetaFlags):
+    INT64 = 0
+    FLOAT16 = 1
+    FLOAT32 = 2
+    FLOAT64 = 3
+
+    @classmethod
+    def get(cls, value: Union[str, int, torch.dtype, Tensor, T]) -> T:  # type: ignore
```
Comment (edgarriba) [inline kornia/constants.py:64]: From torch

Comment (ducha-aiki) [reply kornia/constants.py:64]: @edgarriba it accepts more variants than from torch.

**Comment by ducha-aiki** [comment]
@shijianjian typing tests are still failing. Partly because of new introduced NERF block by @YanivHollander, but augmentation as well...

**Comment by edgarriba** [comment]
shall we merge this ? seems that we need  fixes across modules that can be fixed in posterior prs

**Comment by shijianjian** [comment]
@edgarriba We can. Some errors are not reproduced in my local machine. I will fix it later.

**Pull request merged**